uppsrc/Core/Utf.cpp

```diff
@@ -7,6 +7,17 @@ bool CheckUtf8(const char *s, int len)
 	return FromUtf8_([](wchar) {}, s, len);
 }
 
+int CheckUtf8Split(const char *s, int len)

```

Comment:
Do we really need to add such poorly defined (at the end? Somehow I cannot find real world usage) function just because of test?

Reply:
> Do we really need to add such poorly defined (at the end? Somehow I cannot find real world usage) function just because of test?

I can understand  this, but I am not proposing this just because of a test.  Especially the text mode applications, using;

- ssh or other binary protocols to move data back and forward
- local and remote applications with heavy data IO.

suffers from this frequently. This is more pronounced in text mode applications with high throughput (such as modern terminal apps using ncurses or other text user interface, eg. btop, htop). Almost all modern terminals (including TerminalCtrl), multiplexers and some apps does this check in order to prevent or minimize possible data corruption. Not to mention web apps such as Chromium, Firefox et al have similar helper functions or methods.